proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -736,4 +744,36 @@ public Scoreboard getScoreboard()
     {
         return serverSentScoreboard;
     }
+
+    @Override
+    public void sendResourcePack(String url, String hash, boolean forced, BaseComponent[] promptMessage) throws IllegalArgumentException
+    {
+        if ( hash.isEmpty() )
+        {
+            hash = null;
+        }
+        if ( hash != null )
+        {
+            Preconditions.checkArgument( hash.length() == 40, "Hash must be 40 characters." );
+
+            if ( hash.equalsIgnoreCase( resourcePackHash ) )
+            {
+                return;
+            }
+        }
+
+        if ( bungee.getPluginManager().callEvent( new ResourcePackRequestEvent( this, this, url, hash, forced, promptMessage ) ).isCancelled() )
```

Comment:
Don't think this is needed;
see a few lines below - ```unsafe().sendPacket( new ResourcePackRequest( url, hash, forced, promptMessage ) );``` which will execute the event call in DownstreamBridge

https://github.com/SpigotMC/BungeeCord/pull/3252/files#diff-f5ad33b108028581d9c0580a27e47c53817c7c003b33589d993384a6bee04f54R593

Reply:
My tests conclude that it doesn't call the downstream bridge.

Reply:
downstream bridge is part of the server-side read connection of bungee - when sending a packet to the player it is not involved